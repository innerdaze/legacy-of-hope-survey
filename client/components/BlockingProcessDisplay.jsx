import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Layer from 'grommet/components/Layer'
import Box from 'grommet/components/Box'

class BlockingProcessDisplay extends Component {
  render() {
    return (
        <Layer>
          <Box
            justify='center'
            align='center'
            full={true}>
            {this.props.component}
          </Box>
        </Layer>
    )
  }
}

BlockingProcessDisplay.propTypes = {
  component: PropTypes.object.isRequired
}

BlockingProcessDisplay.defaultProps = {
  component: null
}

export default BlockingProcessDisplay
