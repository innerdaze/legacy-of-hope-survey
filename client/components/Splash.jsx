import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Box from 'grommet/components/Box'
import Label from 'grommet/components/Label'

class Splash extends Component {
  render() {
    return (
      <Box
        full
        justify='center'
        align='center'
        >
        {this.props.icon}
        {this.props.logo}
        {this.props.title}
        <Label>{this.props.loadingText}</Label>
      </Box>
    )
  }
}

Splash.propTypes = {
  loadingText: PropTypes.string,
  icon: PropTypes.instanceOf(Component),
  logo: PropTypes.instanceOf(Component),
  title: PropTypes.instanceOf(Component)
}

Splash.defaultProps = {
  loadingText: 'Loading...',
  icon: null,
  logo: null,
  tite: null
}

export default Splash
