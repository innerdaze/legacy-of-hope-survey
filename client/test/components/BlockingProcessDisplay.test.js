import React from 'react'
import BlockingProcessDisplay from '../../components/BlockingProcessDisplay'
import Splash from '../../components/Splash'

let wrapper, component

describe('Testing on BlockingProcessDisplay...', () => {
  beforeEach(() => {
    component = <Splash />
    wrapper = shallow(<BlockingProcessDisplay component={component} />)
  })

  describe('BlockingProcessDisplay layout', () => {
    test('Expect BlockingProcessDisplay to have Box as parent of component prop that passing in', () => {
      expect(wrapper.find('Box')).toHaveLength(1)
      expect(wrapper.find('Box').find('Splash')).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
