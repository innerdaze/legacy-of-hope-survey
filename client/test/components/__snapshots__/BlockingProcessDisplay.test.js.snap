// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on BlockingProcessDisplay... BlockingProcessDisplay layout Expect BlockingProcessDisplay to have Box as parent of component prop that passing in 1`] = `
<Layer
  align="center"
>
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    full={true}
    justify="center"
    pad="none"
    responsive={true}
    tag="div"
  >
    <Splash
      icon={null}
      loadingText="Loading..."
      logo={null}
      title={null}
    />
  </Box>
</Layer>
`;
